ui/nuxeo-routing-behavior.js

```diff
@@ -107,26 +111,29 @@ export const RoutingBehavior = {
     return function(...args) {
       if (this.router) {
         const [route, ...params] = args;
-        const baseUrl = this.router.baseUrl || '';
         let path;
+        let repo;
         if (typeof route === 'object') {
-          path = this._routeEntity(...args);
+          ({ path, repo } = this._routeEntity(...args));
         } else {
           if (route.startsWith('/')) {
-            this.router.navigate(baseUrl + route);
+            this.router.navigate((this.router.baseUrl || '') + route);
           }
           if (!this.router[route]) {
             console.error(`Could not navigate to a url for route ${route}`);
           }
           path = this.router[route].apply(this, params);
         }
-        this.router.navigate(path);
+        this.router.navigate(path, repo);
```

Comment:
couldn't we just build the right multi repo path here instead of having to make `navigate` be aware of multirepo when before it was just focused on doing the actual navigation?

Reply:
I could not achieve this because `router.navigate` relies on page, and if you pass along the whole multi repo path, it will navigate to a bad URL, something like `/nuxeo/repo/default/ui/#!/nuxeo/repo/second/ui/...`. This means that it is useful to know on `router.navigate` if we are dealing with a repo and if the repository is different then the current one. If we do, we need the full URL and to do specific logic to update the baseUrl on the application side. If we don0t, we use regular `page(...)` navigation.

The alternative would be to compute the whole path and then on `router.navigate`, check if the path starts with any of the known repository hrefs, and if it does, strip down the hashbang and the repo part, if we are doing normal navigation using `page`, but I'm not sure this solution is any better. We could also ditch `page(...)` for navigation, which would probably make this simpler.

Reply:
OK, behaviour is expected because we set the `/repo/<repo>` bit as part of the baseUrl in our index page so the routing system just assumes everything is relative to that and now we need a circuit breaker to not do that.
I'm just wondering if it would make more sense to not set the base Url to the current repo in the first place and have the routing paths explicitly contain the repository. The initial aim was to support multi repository but not cross repository navigation (at least not with the same UI). If this premise has changed perhaps we need to review the choices made at the time.

Reply:
This could probably work if we dropped the baseUrl from page (and the application?) and always computed the full url for both urlFor and naviagateTo. But if someone has a custom router, it would break.